code/game/machinery/bio-dome_floodlights.dm

```diff
@@ -19,6 +19,13 @@
 		F.fswitch = src
 	start_processing()
 
+/obj/structure/machinery/hydro_floodlight_switch/Destroy()
+	for(var/obj/structure/machinery/hydro_floodlight/floodlight as anything in floodlist)
```

Comment:
would QDEL NULL LIST work here

Reply:
i don't think so - you can't do 

`QDEL_NULL_LIST(floodlist.fswitch)` or whatever